heudiconv/dicoms.py

```diff
@@ -203,6 +194,15 @@ def validate_dicom(
     except AttributeError:
         lgr.info("File {} is missing any StudyInstanceUID".format(fl))
         file_studyUID = None
+    if dcmfilter is not None and dcmfilter(mw.dcm_data):
+        lgr.warning("Ignoring %s because of DICOM filter", fl)
+        return None
```

Comment:
I am a little confused -- it seems that you moved filtering **later** not earlier!

I do see that you made it so it clears/deletes signatures now **later** than filtering is done, so would that suffice -- just to move deletion of signatures and keeping the rest of the code as is?